packages/app/src/cli/services/dev/extension/bundler.ts

```diff
@@ -175,33 +178,88 @@ interface SetupConfigWatcherOptions {
   stderr: Writable
   signal: AbortSignal
   unifiedDeployment: boolean
+  configWatcher?: FSWatcher
 }
 
-export async function setupConfigWatcher({
-  extension,
-  token,
-  apiKey,
-  registrationId,
+export async function createConfigWatcher(path: string, onChange: () => void) {
+  const {default: chokidar} = await import('chokidar')
+
+  return chokidar.watch(path).on('change', onChange)
+}
+
+export async function setupConfigExtensionsWatcher({
+  path,
+  extensions,
   stdout,
   stderr,
-  signal,
+  token,
+  apiKey,
   unifiedDeployment,
-}: SetupConfigWatcherOptions) {
-  const {default: chokidar} = await import('chokidar')
+  signal,
+}: {path: string; extensions: ExtensionWithRegistrationId[]} & Omit<
+  SetupConfigWatcherOptions,
+  'extension' | 'registrationId'
+>) {
+  const configWatcher = await createConfigWatcher(path, () => {
+    outputInfo(`Config file at path ${path} changed`, stdout)
 
-  const configWatcher = chokidar.watch(extension.configuration.path).on('change', (_event, _path) => {
-    outputInfo(`Config file at path ${extension.configuration.path} changed`, stdout)
-    updateExtensionConfig({
-      extension,
-      token,
-      apiKey,
-      registrationId,
-      stdout,
-      stderr,
-      unifiedDeployment,
-    }).catch((_: unknown) => {})
+    Promise.all(
+      extensions.map(async (extension) => {
+        return updateExtensionConfig({
+          extension,
+          token,
+          apiKey,
+          registrationId: extension.registrationId,
+          stdout,
+          stderr,
+          unifiedDeployment,
+        })
+      }),
+    ).catch((_: unknown) => {})
   })
 
+  signal.addEventListener('abort', () => {
+    extensions.forEach((extension) => {
+      outputDebug(`Closing config file watching for extension with ID ${extension.devUUID}`, stdout)
+    })
+
+    configWatcher
+      .close()
+      .then(() => {
+        extensions.forEach((extension) => {
+          outputDebug(`Config file watching closed for extension with ${extension.devUUID}`, stdout)
+        })
+      })
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      .catch((error: any) => {
+        extensions.forEach((extension) => {
+          outputDebug(
+            `Config file watching failed to close for extension with ${extension.devUUID}: ${error.message}`,
+            stderr,
+          )
+        })
+      })
+  })
+}
+
+export async function setupConfigWatcher(options: SetupConfigWatcherOptions) {
```

Comment:
This watcher is created on the current app config TOML and can be used to update multiple instances of app config specifications